### PR TITLE
TD-MPC2 Implementation

### PR DESCRIPTION
**Pull request by luizfacury**
This pull request introduces a new training pipeline for the TD-MPC2 world model and policy. The main changes include adding a comprehensive configuration file for TD-MPC2, implementing a new training script with data preprocessing and model management, and registering the TD-MPC2 model in the world model package. These updates enable flexible, modular training of TD-MPC2 using Hydra and PyTorch Lightning, and support multiple observation modalities.

**TD-MPC2 Training Pipeline Integration:**

* Added a new Hydra configuration file `tdmpc2.yaml` specifying all hyperparameters for data loading, model architecture, planning, optimization, and logging for the TD-MPC2 algorithm.
* Implemented `tdmpc2.py`, a new training script that:
  - Dynamically builds datasets and preprocessing transforms for multiple modalities (e.g., pixels, state).
  - Defines the TD-MPC2 forward pass, loss computation, and policy update logic.
  - Sets up PyTorch Lightning training, including a callback for periodic model checkpointing.
  - Integrates with the stable world model and stable pretraining libraries for modular training and data management.

**Model Registration:**

* Registered the TD-MPC2 model and module in the `stable_worldmodel.wm` package by updating the `__init__.py` file, making it available for import and use throughout the codebase.

### CODE REVIEW

File: stable_worldmodel/wm/tdmpc2.py [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+        q_mean = q_values.mean(dim=0)
+        q_std = q_values.std(dim=0)
+
+        penalty = c * q_mean.abs() * q_std
+        conservative_q = q_mean - penalty
+
+        return G + discount * (1 - termination) * conservative_q
+
+    @torch.no_grad()
+    def _plan(self, obs_dict, goal_dict, step_idxs, eval_mode=False):
```
Comment (quentinll) [inline stable_worldmodel/wm/tdmpc2.py]: It would be nice to use the MPPI algorithm already implemented in stable-worldmodel. I believe, one way would be to use the policy API defined here https://github.com/galilai-group/stable-worldmodel/blob/main/stable_worldmodel/policy.py

**Review by quentinll — COMMENTED** [review]
Thank you for this nice contribution! Do you already have results using TDMPC2 in some environments?

**Review by lucas-maes — CHANGES_REQUESTED** [review]

File: stable_worldmodel/wm/tdmpc2.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+                nn.Conv2d(6, 32, 7, stride=2),
+                nn.Mish(),
+                nn.Conv2d(32, 32, 5, stride=2),
+                nn.Mish(),
+                nn.Conv2d(32, 32, 3, stride=2),
+                nn.Mish(),
+                nn.Conv2d(32, 32, 3, stride=1),
+                nn.Mish(),
+                nn.Flatten(),
+            )
```
[the remark this answers is not in the document]
Comment (luizfacury) [reply stable_worldmodel/wm/tdmpc2.py]: It's fine, the cnn works regardless of image size, and output is configured accordingly. I tested different dimensions too
